File: Python/Queue/queue-array.py
class Queue:
    def __init__(self):
        self.items = []

    # Time complexity: O(1)
    def enqueue(self, element):
        self.items.append(element)

    # Time complexity: O(N)
    def dequeue(self):
        if not self.is_empty():
            return self.items.pop(0)
        return None

    # Time complexity: O(1)
    def peek(self):
        if not self.is_empty():
            return self.items[0]
        return None

    # Time complexity: O(1)
    def is_empty(self):
        return len(self.items) == 0

    # Time complexity: O(1)
    def size(self):
        return len(self.items)

    # Time complexity: O(N)
    def print(self):
        print(self.items)

queue = Queue()
print(queue.is_empty())
queue.enqueue(10)
queue.enqueue(20)
queue.enqueue(30)
print(queue.size())
queue.print()
print(queue.dequeue())
print(queue.peek())
queue.print()

File: JavaScript/Queue/queue-array.js
class Queue {
    constructor() {
      this.items = [];
    }
  
    // Time complexity: O(1)
    enqueue(element) {
      this.items.push(element);
    }
  
    // Time complexity: O(N)
    dequeue() {
      return this.items.shift();
    }
  
    // Time complexity: O(1)
    peek() {
      if (!this.isEmpty()) {
        return this.items[0];
      }
      return null;
    }
  
    // Time complexity: O(1)
    isEmpty() {
      return this.items.length === 0;
    }
  
    // Time complexity: O(1)
    size() {
      return this.items.length;
    }
  
    // Time complexity: O(N)
    print() {
      console.log(this.items.toString());
    }
  }
  
  const queue = new Queue();
  console.log(queue.isEmpty());
  queue.enqueue(10);
  queue.enqueue(20);
  queue.enqueue(30);
  console.log(queue.size());
  queue.print();
  console.log(queue.dequeue());
  console.log(queue.peek());
  queue.print();
  